Kernel/LPOComparator.hpp

```diff
@@ -39,43 +39,24 @@ class LPOComparator
     T_UNKNOWN,
   };
 
-  struct Node;
+  class Node;
 
   struct Branch {
     BranchTag tag;
-    Node* n;
+    std::shared_ptr<Node> n;
 
     explicit Branch(BranchTag t) : tag(t), n(nullptr) { ASS(t==BranchTag::T_GREATER || t==BranchTag::T_NOT_GREATER); }
```

Comment:
I think `n(nullptr)` is now the default.

Reply:
I would leave it here to be explicit about storing null in this case.